### PR TITLE
Create FacebookAuthorName component

### LINKED ISSUE #42: Create a FacebookAuthorName component
The `FacebookAuthorName` should be placed within the `FacebookSiteName` component (https://github.com/Yoast/yoast-components/issues/807)
yoast_-_home

When the name is present it should start with a separator (`|`). The name itself should be styled as a link (but it is not a link).

## Styling
The styling for the separator (`span`) is 
```css
    padding-left: 5px;
    padding-right: 5px;
}
```

The styling for the name (link) will be:
```css
    color: #3b5998;
    position: relative;
    text-decoration: none;
    cursor: pointer;
    z-index: 5;
```

Needs https://github.com/Yoast/yoast-components/issues/807

Parent #46

### PR DESCRIPTION
**Pull request by igorschoester**
## Summary

This PR can be summarized in the following changelog entry:

* Adds a FacebookAuthorName component.
* Adds a FacebookSiteAndAuthorNames component.

## Relevant technical choices:

* Instead of placing the `FacebookAuthorName` within the `FacebookSiteName` I created a `FacebookSiteAndAuthorNames` to hold them both.
* Changed `FacebookSiteName`'s HTML element to span because it should be inline.
* Adapted `FacebookPreview` to use `FacebookSiteAndAuthorContainer` instead of `FacebookSiteName`.

## Test instructions

This PR can be tested by following these steps:

* Check the tests.
* Start the stand-alone.
* Go to the FacebookPreview page by clicking on the button in the top right.
* Check if they look according to spec (see issue or better yet Facebook itself).

## UI changes
* [ ] This PR changes the UI in the plugin. I have added the 'UI change' label to this PR.

Fixes #42

### CODE REVIEW

File: packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import React from "react";
```
Comment (afercia) [inline packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js]: I'd add the external dependencies comment.

File: packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import styled from "styled-components";
+import PropTypes from "prop-types";
+
+const FacebookAuthorNameWrapper = styled.span`
+    color: #3b5998;
```
Comment (afercia) [inline packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js]: Ideally, colors should be in the color palette. Not sure in this case, as this is a single-use color.

File: packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    font-size: 12px;
+    line-height: 11px;
```
Comment (afercia) [inline packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js]: Any reason why the line-height is so small? Also, ideally it should be unitless. I'd go for `line-height: 1;`

File: packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    position: relative;
```
Comment (afercia) [inline packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js]: position: relative; doesn't do anything here, it should be removed.

File: packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    text-decoration: none;
```
Comment (afercia) [inline packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js]: text-decoration: none; doesn't do anything: this is not a link and it's not underlined by default. It should be removed.

File: packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    cursor: auto;
```
Comment (afercia) [inline packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js]: cursor: auto; doesn't do anything: this is not a link, just remove it please.

File: packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+`;
```
Comment (afercia) [inline packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js]: all the indentation in this ruleset is made with spaces, should be changed to tabs.

File: packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookSiteAndAuthorNames.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
+import React, { Fragment } from "react";
+import styled from "styled-components";
+import PropTypes from "prop-types";
+import { __ } from "@wordpress/i18n";
+
+import FacebookSiteName from "./FacebookSiteName";
+import FacebookAuthorName from "./FacebookAuthorName";
+
+const FacebookSiteAndAuthorNamesWrapper = styled.p`
+    color: #606770;
```
Comment (afercia) [inline packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookSiteAndAuthorNames.js]: same as above about colors

File: packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookSiteAndAuthorNames.js [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    font-size: 12px;
+    line-height: 11px;
```
Comment (afercia) [inline packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookSiteAndAuthorNames.js]: same as above about line-height

File: packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookSiteAndAuthorNames.js [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+    overflow: hidden;
+    text-overflow: ellipsis;
+    text-transform: uppercase;
+    white-space: nowrap;
+`;
```
Comment (afercia) [inline packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookSiteAndAuthorNames.js]: same as above: all this ruleset uses spaces for indentation, should be changed to tabs

File: packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookSiteAndAuthorNames.js [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+
+const FacebookSiteAndAuthorNamesSeparator = styled.span`
+    padding-left: 5px;
+    padding-right: 5px;
+`;
```
Comment (afercia) [inline packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookSiteAndAuthorNames.js]: same as above: all this ruleset uses spaces for indentation, should be changed to tabs

File: packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookSiteName.js [hunk 12]
```diff
@@ -2,7 +2,7 @@ import React from "react";
 import styled from "styled-components";
 import PropTypes from "prop-types";
 
-const FacebookSiteNameWrapper = styled.p`
+const FacebookSiteNameWrapper = styled.span`
```
Comment (afercia) [inline packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookSiteName.js]: whit this change, some of the CSS rules should be removed.
text-overflow: ellipsis; doesn't work on inline elements
as a consequence, white-space: nowrap; should be removed as well (the on on the container will work)
re: the line-height and color, see previous considerations

**Comment by afercia** [comment]
CR 🚧 

See comments, mainly CSS stuff.

File: packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js (hunk 1 above)
Comment (johannadevos) [reply packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js]: Done

File: packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookSiteAndAuthorNames.js [hunk 13]
```diff
@@ -0,0 +1,75 @@
+import React, { Fragment } from "react";
```
[the remark this answers is not in the document]
Comment (johannadevos) [reply packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookSiteAndAuthorNames.js]: Done

File: packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js (hunk 2 above)
Comment (johannadevos) [reply packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js]: @afercia What do you mean with "the color palette"?

Comment (afercia) [reply packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js]: @johannadevos a while ago we've built a color palette to control all the colors used in the components. Then in the components we should use the variables representing the colors. See 
https://github.com/Yoast/javascript/blob/develop/packages/yoast-components/style-guide/_colors.scss which then gets compiled in a [json file](https://github.com/Yoast/javascript/blob/develop/packages/yoast-components/style-guide/colors.json) for reuse in JavaScript. 
In this specific case I'd agree with @igorschoester: the Facebook and Twitter colors aren't part of our design so maybe we can make an exception.

Comment (johannadevos) [reply packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js]: @afercia Thanks for the explanation! @igorschoester and @IreneStr both agree that we should make an exception for Facebook and Twitter colours :)

File: packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js (hunk 3 above)
Comment (johannadevos) [reply packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js]: The reason is that we're using the exact same CSS settings as Facebook, and they use `line-hight: 11px;`.

File: packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookSiteAndAuthorNames.js (hunk 9 above)
Comment (johannadevos) [reply packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookSiteAndAuthorNames.js]: The reason is that we're using the exact same CSS settings as Facebook, and they use `line-hight: 11px;`.

File: packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js (hunk 4 above)
Comment (johannadevos) [reply packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js]: Done

File: packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js (hunk 5 above)
Comment (johannadevos) [reply packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js]: Done

File: packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js (hunk 6 above)
Comment (johannadevos) [reply packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js]: Done

File: packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookSiteName.js (hunk 12 above)
Comment (johannadevos) [reply packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookSiteName.js]: Done

File: packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js (hunk 7 above)
Comment (johannadevos) [reply packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookAuthorName.js]: Done

File: packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookSiteAndAuthorNames.js (hunk 10 above)
Comment (johannadevos) [reply packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookSiteAndAuthorNames.js]: Done

File: packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookSiteAndAuthorNames.js (hunk 11 above)
Comment (johannadevos) [reply packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookSiteAndAuthorNames.js]: Done

File: packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookSiteAndAuthorNames.js (hunk 8 above)
Comment (johannadevos) [reply packages/yoast-components/composites/Plugin/SocialPreviews/Facebook/components/FacebookSiteAndAuthorNames.js]: See above

**Comment by IreneStr** [comment]
CR 👍

**Comment by IreneStr** [comment]
Acceptance 👍 
I've created an issue to address some screenreader concerns: https://github.com/Yoast/javascript/issues/89

**Pull request merged**